traffic_monitor/experimental/common/poller/poller.go

```diff
@@ -157,68 +187,151 @@ func (p HttpPoller) Poll() {
 				fetcher.Client = &c // copy the client, so we don't change other fetchers.
 				fetcher.Client.Timeout = info.Timeout
 			}
-			go pollHttp(info.Interval, info.ID, info.URL, fetcher, kill)
+			go sleepPoller(info.Interval, info.ID, info.URL, fetcher, kill)
 		}
 		p.Config = newConfig
 	}
 }
 
-type HTTPPollInfo struct {
-	Interval time.Duration
-	Timeout  time.Duration
-	ID       string
-	URL      string
-	Handler  handler.Handler
+func mustDie(die <-chan struct{}) bool {
+	select {
+	case <-die:
+		return true
+	default:
+	}
+	return false
 }
 
-// diffConfigs takes the old and new configs, and returns a list of deleted IDs, and a list of new polls to do
-func diffConfigs(old HttpPollerConfig, new HttpPollerConfig) ([]string, []HTTPPollInfo) {
-	deletions := []string{}
-	additions := []HTTPPollInfo{}
+// TODO iterationCount and/or p.TickChan?
+func sleepPoller(interval time.Duration, id string, url string, fetcher fetcher.Fetcher, die <-chan struct{}) {
+	pollSpread := time.Duration(rand.Float64()*float64(interval/time.Nanosecond)) * time.Nanosecond
+	time.Sleep(pollSpread)
+	tick := time.NewTicker(interval)
+	lastTime := time.Now()
+	for {
+		select {
+		case <-tick.C:
+			realInterval := time.Now().Sub(lastTime)
+			if realInterval > interval+(time.Millisecond*100) {
+				instr.TimerFail.Inc()
+				log.Infof("Intended Duration: %v Actual Duration: %v\n", interval, realInterval)
```

Comment:
I was trying to avoid filling up the log with information that is not useful, but if it is useful we should keep it as info.